### PR TITLE
feat(callbox): scaffold component

### PR DESCRIPTION
**Pull request by juliodialpad**
# Feat (Callbox): Scaffold component

## :hammer_and_wrench: Type Of Change

- [ ] Fix
- [x] Feature
- [ ] Refactoring
- [ ] Documentation

## :book: Description

- Created Callbox recipe component
- Used item-layout as base
- As designs were close enough, did some work on advance, it's still not perfect and might change on a different PR.
- Added `no-visual-tests` label as this is not ready for visual testing.

## :bulb: Context

https://dialpad.atlassian.net/browse/DLT-497

## :pencil: Checklist

- [x] I have reviewed my changes
- [ ] I have added tests
- [ ] I have added all relevant documentation
- [ ] I have validated components with a screen reader
- [ ] I have validated components keyboard navigation
- [ ] I have considered the performance impact of my change
- [ ] I have checked that my change did not significantly increase bundle size
- [x] I am exporting any new components or constants in the index.js in the component directory
- [x] I am exporting any new components or constants in the index.js in the root

## :crystal_ball: Next Steps

- Write documentation
- Write tests
- Finish visuals

## :camera: Screenshots / GIFs

image
image
image
image

### CODE REVIEW

**Comment by iropolo** [comment]
As far as I can see It is ok and ready to go with this base.

Just a question, should width be mofified if for example you add more buttons in the right slot?

**Review by iropolo — APPROVED** [review]

**Comment by iropolo** [comment]
There is a slight design difference on spacing, probably due button padding.

image

Screenshot 2023-08-23 at 15 02 34

File: recipes/leftbar/callbox/callbox_default.story.vue [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            </template>
+          </dt-button>
+        </template>
+        <template #default>
+          <span class="d-fs-100">Sharing screen</span>
+        </template>
+        <template #right>
+          <dt-button
+            importance="clear"
+            circle
```
Comment (braddialpad) [inline recipes/leftbar/callbox/callbox_default.story.vue]: remove circle since this contains text

**Comment by iropolo** [comment]
If you provide a non-working image its breaks as this way

image

**Comment by juliodialpad** [comment]
> If you provide a non-working image its breaks as this way
> 
> image

Did you provide a full-name? if you provide a image-src you must provide a full-name for the avatar to be able to display initials instead of the image.

**Comment by juliodialpad** [comment]
> There is a slight design difference on spacing, probably due button padding.
> 
> image Screenshot 2023-08-23 at 15 02 34

This was just a quick example to prove the slots could handle all the possible states, I'll fix that in a later PR.

**Comment by iropolo** [comment]
> > If you provide a non-working image its breaks as this way
> > image
> 
> Did you provide a full-name? if you provide a image-src you must provide a full-name for the avatar to be able to display initials instead of the image.

Ye, seems like it is some reactive issue, if you update the imageSrc from the controls you will see it, but Im guessing it is possible because we are not handle that in the avatar component

File: recipes/leftbar/callbox/callbox.vue [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    },
+
+    /**
+     * Optional avatar image url.
+     * If not provided it will use the initial of the name.
+     */
+    avatarSrc: {
+      type: String,
+      default: '',
+    },
```
Comment (braddialpad) [inline recipes/leftbar/callbox/callbox.vue:99]: probably need to also pass seed or initials background color will constantly change.

**Comment by juliodialpad** [comment]
> > > If you provide a non-working image its breaks as this way
> > > image
> > 
> > 
> > Did you provide a full-name? if you provide a image-src you must provide a full-name for the avatar to be able to display initials instead of the image.
> 
> Ye, seems like it is some reactive issue, if you update the imageSrc from the controls you will see it, but Im guessing it is possible because we are not handle that in the avatar component

It should be handled, I'll check it and fix it. Thanks for catching that.

**Review by braddialpad — APPROVED** [review]
Obviously not fully complete yet but this looks like a pretty great start!

File: recipes/leftbar/callbox/callbox_default.story.vue (hunk 1 above)
Comment (juliodialpad) [reply recipes/leftbar/callbox/callbox_default.story.vue]: Right, fixed

File: recipes/leftbar/callbox/callbox.vue (hunk 2 above)
Comment (juliodialpad) [reply recipes/leftbar/callbox/callbox.vue:99]: Added

**Pull request merged**